### DISCUSSION TITLE
How to Set Default Number Setting for GUI?

### DISCUSSION THREAD

**Discussion by utkonos** [Vector35/binaryninja-api#5006] [Q&A] [answered]
I am using the following configuration for one settings entry:

```python
import json

import from binaryninja.settings import Settings

s = Settings()
s.register_group('foo', 'Foo Bar')
setting = {
    'description': 'Enter function count limit.',
    'title': 'Function count limit',
    'default': 10000,
    'type': 'number'
}
s.register_setting('foo.bar', json.dumps(setting))

print(s.get_integer('foo.bar')
```

The default value of 10000 is there. It is printed at the end of that. However, when one looks at the Setting itself in the GUI, it is displayed as zero:

<img width="234" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/50291090-0acb-4fb7-916d-f59160d52499">

**Comment by fuzyll** [comment, marked as answer]
That looks right to me? Are you able to also set `minValue` to `0` and `maxValue` to a larger number? It's possible the default `maxValue` is preventing you from specifying 10000.

**Reply by utkonos** [reply]
Yes, that's it. Thank you.

```python
setting = {
    'description': 'Enter function count limit.',
    'title': 'Function count limit',
    'default': 10000,
    'minValue': 10000,
    'type': 'number'
}
```

<img width="202" alt="image" src="https://github.com/Vector35/binaryninja-api/assets/2775739/3108f07b-a516-4752-82b0-6e59bd6ff9df">
